### ISSUE TITLE
[Bug] When more than 1 is used for num_beams : probability tensor contains either `inf`, `nan` or element < 0 

### ISSUE THREAD

**Issue by FurkanGozukara** [coqui-ai/TTS#3232]
### Describe the bug

I am testing num_beams and when more than 1 is used I am getting this error

![image](https://github.com/coqui-ai/TTS/assets/19240467/7a83ab02-2ef6-4cd2-946b-e53bede8f9fa)

Everything is latest version. Here below my all config parameters

![image](https://github.com/coqui-ai/TTS/assets/19240467/d0c6daa6-0bb4-4783-a998-260494f27237)

### Environment

```shell
(venv) F:\coqui_test\TTS\TTS\bin>python collect_env_info.py
{
    "CUDA": {
        "GPU": [
            "NVIDIA GeForce RTX 3090 Ti",
            "NVIDIA GeForce RTX 3060"
        ],
        "available": true,
        "version": "12.1"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.1.0+cu121",
        "TTS": "0.20.5",
        "numpy": "1.22.0"
    },
    "System": {
        "OS": "Windows",
        "architecture": [
            "64bit",
            "WindowsPE"
        ],
        "processor": "Intel64 Family 6 Model 183 Stepping 1, GenuineIntel",
        "python": "3.10.11",
        "version": "10.0.19045"
    }
}
```

**Comment by WeberJulian** [comment]
Yeah, for now it's not intended as controllable, you should leave it to 1. I'll send a PR to remove the arg.

**Comment by WeberJulian** [comment]
My bad I misread it as `num_return_sequences`, according to [this comment](https://github.com/vllm-project/vllm/issues/641#issuecomment-1738840454), this is expected behaviour when temperature is too low and there is not enough candidate. I could reproduce and it's fixed by setting `temperature=1.0`

**Comment by FurkanGozukara** [comment]
> My bad I misread it as `num_return_sequences`, according to [this comment](https://github.com/vllm-project/vllm/issues/641#issuecomment-1738840454), this is expected behaviour when temperature is too low and there is not enough candidate. I could reproduce and it's fixed by setting `temperature=1.0`

thanks so temp should be min 1? if we make it 2 what happens?
